libs/mails/src/lib/mail-provider/mailchimp-mail-provider.ts

```diff
@@ -51,7 +51,14 @@ function mapMandrillEventToMailLogState(event: string): MailLogState | null {
 function flattenObj(ob: any) {
   const nestedObject: any = {}
   for (const i in ob) {
-    if (typeof ob[i] === 'object' && !Array.isArray(ob[i])) {
+    if (Array.isArray(ob[i])) {
```

Comment:
This creates:

``` 
{name: 'user_roleIDs_0_0', content: 'h'} 
{name: 'user_roleIDs_0_1', content: 'i'} 
{name: 'user_roleIDs_0_2', content: 'd'} 
{name: 'user_roleIDs_0_3', content: 'd'}
{name: 'user_roleIDs_0_4', content: 'e'}
{name: 'user_roleIDs_0_5', content: 'n'}
```

Instead of:

```
{name: 'user_roleIDs_0', content: 'hidden'} 
```

This isn't intended, right? @elias-summermatter / @michael-scheurer

Reply:
Your right thats not intended....